src/pages/[platform]/build-a-backend/troubleshooting/library-not-configured/index.mdx

```diff
@@ -0,0 +1,148 @@
+        'angular',
+        'react',
+        'react-native',
+        'vue',
+        'nextjs',
+        'javascript'
+      ],
+      canonicalPath: '/angular/build-a-backend/troubleshooting/migrate-from-javascript-v5-to-v6/'
+    }
+  ]
+};
+
+export const getStaticPaths = async () => {
+  return getCustomStaticPath(meta.platforms);
+};
+
+export function getStaticProps(context) {
+  return {
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+If you are running into the `Library Not Configured` error message and have called `Amplify.configure` in your project, your Amplify API is most likely being called before `Amplify.configure`. This can happen in a few different ways. Below are three possibilities you can check to troubleshoot this issue.
+
+## Call `Amplify.configure` in the root of your project
+
+Make sure you are calling `Amplify.configure` in the root file of your project. The root file of your app may be different depending on your frontend framework. The current default for some common frameworks are listed below (if you are not using TypeScript the `ts` and `tsx` extensions would be `js` and `jsx`):
+
+* Vue.js: **src/main.ts**
+* React: **src/main.tsx**
+* Angular: **src/main.ts**
+* Next.js Page Router: **pages/_app.tsx** or **src/pages/_app.tsx**
+
+<Callout>
+
+If you are using the Next.js App Router, you can follow the suggestions in our [Next.js documentation](/[platform]/build-a-backend/server-side-rendering/nextjs/#configure-amplify-library-for-client-side-usage) for root-level configuration. Keep in mind that if you are calling any API's at the module-level (i.e. at the top of your file) in any of the Child components, you may still run into this issue.
+
+</Callout>
+
+## Check for module-level Amplify API invocations
+
+When Amplify APIs are used outside of your application lifecycle, there is a risk that a javascript bundler may place that API call before `Amplify.configure`. Module-level function calls (calls at the top-level of a file), are generally evaluated in the order that they are imported.
+
+Below is an example of code that will likely result in a `Library Not Configured` error:
+
+```javascript title="index.ts"
+import { Amplify } from 'aws-amplify';
+import ComponentX from 'module-fetch-auth';
+
+// fetchAuthSession() in ComponentX executed on import
+
+Amplify.configure();
+
+export default function App() {
+  return (
+    <div>
+        <ComponentX />
+    </div>
+  );
+}
+```
+
+```javascript title="module-fetch-auth.ts"
+import { fetchAuthSession } from 'aws-amplify/auth';
+
+fetchAuthSession(); // Will throw the Library Not Configured error
+
+export default function ComponentX() {
+  return (
+    <div className="box">
+      ...
+    </div>
+  );
+}
+```
+
+This can also happen when using Layouts (as in NextJS) and calling Amplify API’s in child components at the module-level (at the top of your file/module). See below for an example of this issue:
```

Comment:
```suggestion
This can also happen when using Next.js Layouts and calling Amplify API’s in child components at the module-level (at the top of your file/module). See below for an example of this issue:
```